src/app/AiPlayer.ts

```diff
@@ -0,0 +1,195 @@
+import { AnimalNames } from '~src/Enums/AnimalNamesEnum';
+import { AnimalRoles } from '~src/Enums/AnimalRolesEnum';
+import { Player } from '../Player';
+import { GameController } from './GameController';
+import { Offer } from './Trade';
+import { ConvertAnimalName } from './utils/ConvertAnimalName';
+
+export class AiPlayer extends Player {
+  /**
+   * Makes a move for AI Player
+   * @param gameController for the game that the player is participating in
+   */
+  makeAMove(gameController: GameController): void {
+    this.trade(gameController);
+    if (gameController.theGameProcessor.checkWin()) {
+      gameController.checkIfGameIsWon();
+      return;
+    }
+    gameController.breed();
+    if (gameController.theGameProcessor.checkWin()) {
+      gameController.checkIfGameIsWon();
+      return;
+    }
+  }
+  private trade(gameController: GameController): void {
+    //Try to win the game by trading one to many if possible
+    const oneToManyTrades: [AnimalNames, Offer[]][] = [
+      [
+        AnimalNames.HORSE,
+        [
+          [AnimalNames.COW, 1],
+          [AnimalNames.PIG, 1],
+          [AnimalNames.SHEEP, 3],
+          [AnimalNames.RABBIT, 6],
+        ],
+      ],
+      [
+        AnimalNames.COW,
+        [
+          [AnimalNames.PIG, 1],
+          [AnimalNames.SHEEP, 3],
+          [AnimalNames.RABBIT, 6],
+        ],
+      ],
+      [
+        AnimalNames.PIG,
+        [
+          [AnimalNames.SHEEP, 1],
+          [AnimalNames.RABBIT, 6],
+        ],
+      ],
+      [AnimalNames.SHEEP, [[AnimalNames.RABBIT, 6]]],
```

Comment:
These calculations should be based on the configuration. If any value changes, the calculations go wrong. Please check if you can think of a way to do these calculations using values from config.

Reply:
I totally agree with you. Supporting different values for animals makes the AI Player solution far more complex. I suggest we leave it as is and the AI Player will work only with default configuration when it comes to animal values.